haystack/preview/components/generators/_helpers.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def enforce_token_limit(prompt: str, tokenizer, max_tokens_limit: int) -> str:
```

Comment:
We should probably add a type hint for `tokenizer`.

Reply:
Reflecting a bit more on this type made me realize that this method in fact is not generic at all... I don't think any other LLM other than OpenAI's can use tiktoken for tokenization. So I'll move this method into `generators.openai` rather than leaving it here, and then type it correctly :slightly_smiling_face: